ingestion-beam/src/main/java/com/mozilla/telemetry/util/Normalize.java

```diff
@@ -39,4 +65,64 @@ public static String channel(String name) {
       return OTHER;
     }
   }
+
+  /**
+   * Normalize an operating system string to one of the three major desktop platforms,
+   * one of the two major mobile platforms, or "Other".
+   *
+   * <p>Reimplements and combines the logic of {@code os} and {@code mobile_os} from
+   * https://github.com/mozilla-services/lua_sandbox_extensions/blob/607b3c7b9def600ddbd373a14f67a55e1c69f7a1/moz_telemetry/modules/moz_telemetry/normalize.lua#L184-L215
+   */
+  public static String os(String name) {
+    name = Strings.nullToEmpty(name);
+    if (name.startsWith("Windows") || name.startsWith("WINNT")) {
+      return WINDOWS;
+    } else if (name.startsWith("Darwin")) {
+      return MAC;
+    } else if (name.contains("Linux") || name.contains("BSD") || name.contains("SunOS")) {
+      return LINUX;
+    } else if (name.startsWith("iOS") || name.contains("iPhone")) {
+      return IOS;
+    } else if (name.startsWith("Android")) {
+      return ANDROID;
+    } else {
+      return OTHER;
+    }
+  }
+
+  /**
+   * Normalize an operating system version string to at most MAJOR.MINOR.PATCH numeric version.
+   *
+   * <p>Strips extra text at the end of the version string, but returns an empty string if the
+   * input starts with non-numeric text.
+   *
+   * <p>Reimplementation of https://github.com/mozilla-services/lua_sandbox_extensions/blob/607b3c7b9def600ddbd373a14f67a55e1c69f7a1/moz_telemetry/modules/moz_telemetry/normalize.lua#L196-L204
+   */
+  public static String osVersion(String v) {
+    v = Strings.nullToEmpty(v);
+    Matcher matcher = OS_VERSION_RE.matcher(v);
+    if (matcher.matches() && matcher.groupCount() == 1) {
+      return matcher.group(1);
+    } else {
+      return "";
+    }
+  }
+
+  /**
+   * Normalize app_name as reported by the various products, or "Other" for unrecognized prefixes.
+   *
+   * <p>Reimplements {@code mobile_app_name} from
+   * https://github.com/mozilla-services/lua_sandbox_extensions/blob/607b3c7b9def600ddbd373a14f67a55e1c69f7a1/moz_telemetry/modules/moz_telemetry/normalize.lua#L218-L230
+   * and adds a case for "Firefox" desktop app.
```

Comment:
Likewise here, all tables loaded by ingestion will have a `normalized_app_name` field, so we support the desktop case along with mobile.